packages/e2e/lib/common.dart

```diff
@@ -30,19 +37,34 @@ class Response {
   List<Failure> get failureDetails => _failureDetails;
 
   /// Serializes this message to a JSON map.
-  String toJson() => json.encode(<String, dynamic>{
-        'result': allTestsPassed.toString(),
-        'failureDetails': _failureDetailsAsString(),
+  String toJson() {
+    final Map<String, dynamic> result = <String, dynamic>{};
+    result['result'] = allTestsPassed.toString();
+    result['failureDetails'] = _failureDetailsAsString();
+    if (extraInfo != null) {
+      extraInfo.forEach((String key, dynamic value) {
```

Comment:
Why don't we just put this in an additional key in the map, to avoid overwriting or special casing any key names?

Reply:
I'm feeling `extraInfo` as a variable name is fine but as a key in json it's too colloquial. Maybe I should call it `report`.

Reply:
How about just 'data'?

Reply:
Yeah that's simpler. Done